contracts/utils/math/Math.sol

```diff
@@ -76,15 +76,32 @@ library Math {
     /**
      * @dev Returns the largest of two numbers.
      */
-    function max(uint256 a, uint256 b) internal pure returns (uint256) {
-        return a > b ? a : b;
+    function max(uint256 a, uint256 b) internal pure returns (uint256 result) {
+        return choice(a > b, a, b);
     }
 
     /**
      * @dev Returns the smallest of two numbers.
      */
-    function min(uint256 a, uint256 b) internal pure returns (uint256) {
-        return a < b ? a : b;
+    function min(uint256 a, uint256 b) internal pure returns (uint256 result) {
+        return choice(a < b, a, b);
+    }
+
+    /**
+     * @dev If `condition` is true, returns `a`, otherwise returns `b`.
+     */
+    function choice(bool condition, uint256 a, uint256 b) internal pure returns (uint256) {
```

Comment:
I really like this function. 

I would rename that `ternary(bool, uint256, uint256)` though. @ernestognw, WDYT?

Reply:
Another suggestion is using the name `select`, this is the name used in [openssl constant time library](https://github.com/openssl/openssl/blob/openssl-3.2.0/include/internal/constant_time.h#L276-L281).

Reply:
I like both but I feel like `ternary` is more explicit than `select`.

Note that Solidity is also frequently used by backend developers who may be used to the meaning of `select` in SQL. My opinion is that `ternary` is better scoped to what it does and its arguments follow the same order as in a ternary (e.g. `a ? b : c` vs `ternary(a, b, c)`)

Reply:
Ok, will change to `ternary` then.

Reply:
done